src/it/resources/com/google/checkstyle/test/chapter4formatting/rule4861blockcommentstyle/InputCommentsIndentationCommentIsAtTheEndOfBlock.java

```diff
@@ -15,7 +15,7 @@ public void foo1() {
 
     public void foo2() {
         foo3();
-                         // warn
+                         // violation 'Comment has incorrect indentation level 25, expected is 8, indentation should be the same level as line 17.'
```

Comment:
`// violation 'indentation should be the same level as line 17.'`
and can you place this before foo3 with collect indentation.
so it will be:
``// violation 2 lines below, 'indentation should be the same level as line 17.'``

apply the same  trick to all other cases.

`// warn` ===> `// odd indentation comment`

Reply:
I tried this. I have tried to keep the `// violation above/below` near to the code line which defines the indentation level and replaces the `//warn` with `// odd.....` as you suggested.